scrapers/sis_scraper/main.py

```diff
@@ -1,546 +1,306 @@
-"""
-            )
-
-            for crn, conflict in conflicts.items():
-                rust_array = f"\t{crn}u32 => ["
-                for i in range(0, BIT_VEC_SIZE * 64, 64):
-                    if i != 0:
-                        rust_array += ", "
-                    rust_array += str(int(conflict[i : i + 64], 2))
-                rust_array += "],\n"
-
-                f.write(rust_array)
-
-            f.write(
-                """
-};
-
-pub static CRN_COURSES: Map<u32, &'static str> = phf_map! {
-"""
+        res_school = {"name": possible_school["name"], "depts": []}
+        for target_dept in possible_school["depts"]:
+            matching_depts = list(
+                filter(lambda d: d["code"] == target_dept["code"], courses)
             )
+            if matching_depts:
+                res_school["depts"].append(target_dept)
+        if res_school["depts"]:
+            matched_subjects.update(d["code"] for d in res_school["depts"])
+            schools.append(res_school)
+    # Determine if any department is missing from schools.json list and
+    # put missing ones into an "Uncategorized" school. This has happened a few times in the past,
+    # most notably when STSH and STSS merged to become STSO.
+    all_subjects = set(d["code"] for d in courses)
+    unmatched_subjects = all_subjects - matched_subjects
+    schools.append(
+        {
+            "name": "Uncategorized",
+            "depts": list(
+                {
+                    "code": code,
+                    "name": list(filter(lambda dept: dept["code"] == code, courses))[0][
+                        "name"
+                    ],
+                }
+                for code in unmatched_subjects
+            ),
+        }
+    )
 
-            for crn, course in crn_to_courses.items():
-                f.write(f'\t{crn}u32 => "{course}",\n')
-            f.write("};")
+    school_columns = util.optimize_column_ordering(schools)
+    # Write out all the results of the scraper
+    conflict_logic.gen(term, courses)
+    # Replace all the dateStart/dateEnd with the MM/DD format used by the quacs frontend
+    # Additionally, split out the prereq field into a separate json
+    prerequisites = {}
+    date_to_quacs = lambda date: f"{str(date.month).zfill(2)}/{str(date.day).zfill(2)}"
+    for dept in courses:
+        for course in dept["courses"]:
+            for section in course["sections"]:
+                try:
+                    prerequisites[section["crn"]] = section["prereqs"]
+                    del section["prereqs"]
+                except:
+                    pass
+                for timeslot in section["timeslots"]:
+                    timeslot["dateStart"] = date_to_quacs(timeslot["dateStart"])
+                    timeslot["dateEnd"] = date_to_quacs(timeslot["dateEnd"])
+    with open(f"data/{term}/schools.json", "w") as schools_f:
+        json.dump(school_columns, schools_f, sort_keys=False, indent=2)
```

Comment:
```suggestion
        json.dump(school_columns, schools_f, sort_keys=True, indent=2)
```

Given the diff is not going to match anyway, I think now is as good a time as any to sort the keys so it stays more consistent in the long run

Reply:
updated https://github.com/quacs/quacs/pull/690/commits/906eaf6f9c0b03dce98335f22513e25a3fadce6e

note: i left schools.json alone because that is manually sorted by the algorithm to optimize school layout.